### DISCUSSION TITLE
Minimal example for running workflows via act without github?

### DISCUSSION THREAD

**Discussion by kreha1** [ChristopherHX/github-act-runner#83] [Q&A] [answered]
Hi, I was trying to get it running with your version of act, but it won't recognize self-hosted runner and wants me to select image size, which then modifies `.actrc` file. It always wants me to set docker image for the runner.

I'm using GHE at work and I wanted to improve time debugging workflows in a big monorepo. That is how I found your project after act didn't work for me as it only provides virtual environments which we don't use.

**Comment by ChristopherHX** [comment, marked as answer]
You mean this component https://github.com/ChristopherHX/act?
It has an undocumented flag to run on your host machine for example
```yaml
on: push
jobs:
  _:
    runs-on: self-hosted
    steps:
    - run: echo "Run without docker"
```
Run act like
```
act -P self-hosted=-self-hosted
```

_I have planned that this mode will be integrated in nektos/act, https://github.com/nektos/act/pull/1293_

I prefer to use actions/runner to run things locally on supported platforms and created this drop in replacement in 2021 for act https://github.com/ChristopherHX/runner.server/, this programm understands the flag `-P self-hosted=-self-hosted` too.

**Reply by kreha1** [reply]
I had actually read other components you've mentioned in this repo's readme file, and I had tried using [this example .actrc configuration](https://github.com/ChristopherHX/runner.server/#the-actrc-file), just the `self-hosted` bit like you've mentioned in the reply... and it turns out that the `linux` part in `-P self-hosted,linux=-self-hosted` was marking it as unsupported platform (even though I'm on ubuntu 18.04).
So thanks for the quick reply, as it helped :).

I'm wondering now if I run `runner.server` in a container and expose its port, should your version of act still recognize it?

**Reply by ChristopherHX** [reply]
The instructions of `runner.server` doesn't work with `act`, runner.server is my github actions server software for self-hosting github actions without using github-enterprise. `runner.server` has a cli client named `Runner.Client`/`gharun`, which supports this syntax `-P self-hosted,linux=-self-hosted `

My version of act only understands `-P self-hosted=-self-hosted -P linux=-self-hosted` and it doesn't use anything from `runner.server` at all.

> I'm wondering now if I run runner.server in a container and expose its port, should your version of act still recognize it?

I think my documentation confused you a bit, you cannot use [ChristopherHX/act](https://github.com/ChristopherHX/act) with [ChristopherHX/runner.server](https://github.com/ChristopherHX/runner.server) these programs are not compatible.

_However you can use the binary `github-act-runner` from this repository as a self-hosted runner and configure it against github.com or [ChristopherHX/runner.server](https://github.com/ChristopherHX/runner.server)_

**Reply by kreha1** [reply]
I see. But again, thank you. Now I can start playing with it, and I hope that it will greatly decrease iteration time for debugging my workflows :).
